src/einsteinpy/symbolic/tensor.py

```diff
@@ -453,3 +469,52 @@ def lorentz_transform(self, transformation_matrix):
             functions=self.functions,
             name=_change_name(self.name, context="__lt"),
         )
+
+    def derivative(self, x):
+        """Ordinary Derivative of a tensor along the direction given by x
+
+        Parameters
+        ----------
+        x : ~einsteinpy.symbolic.BaseRelativityTensor or ~sympy.Symbol or list
+            differentiate along vector x
+
+            Assumes, if ``x`` is of type:
+            list: vector components
+            symbols: a coordinate vector
+            BaseRelativityTensor: a tensor of config 'u'
+
+        Returns
+        -------
+        ~einsteinpy.symbolic.BaseRelativityTensor
+
+        Raises
+        ------
+        ValueError
+            If ``x`` is a list but length doesn't match length of ``syms``
+        TypeError
+            If argument ``x`` is of type other than listed above
+        """
+        if (
+            isinstance(x, BaseRelativityTensor)
+            and x.config != "u"
+            and x.syms != self.syms
+        ):
+            raise ValueError("Expected a tensor of config 'u' with syms %s" % self.syms)
+        if not isinstance(x, BaseRelativityTensor):
+            x = _parse_vector(x, self.syms)
+
+        arr = np.zeros(self.arr.shape, dtype=int)
+        arr = sympy.Array(arr)
+        for i in range(len(self.syms)):
+            if x[i]:
+                arr += x[i] * self.arr.diff(self.syms[i])
+
```

Comment:
- I think this code would fail if you pass as an argument a list of one symbol, e.g :
```
obj1.derivative([x])

```
Maybe you should test it in order to check if fails.

- Moreover, this code will work one variable derivative, but have you contemplate the option of operating with multivariable-derivative (e.g.  ∂/(∂x ∂y))?

Reply:
>     * I think this code would fail if you pass as an argument a list of one symbol, e.g :
> 
> 
> ```
> obj1.derivative([x])
> ```

This is true. For one symbol argument, one could simply pass in the symbol. 
`obj1.derivative(x)`

>    (e.g.  ∂/(∂x ∂y))?

By, this did you mean second partial derivative? If so, you'd call `.derivative()` twice:
`obj1.derivative(y).derivative(x)`

If you mean like directional derivative along certain direction other than coordinate axes, that's when you'd want to pass a `BaseRelativityTensor`.  Isn't that right ?

Reply:
Yes, by doing '.derivative' twice works. And this can be also a possible test to indicate how to make second partial derivative (instead of .derivate(x,y) or similar)

Reply:
And sure "For one symbol argument, one could simply pass in the symbol", but it is better to know when it can fail.

Reply:
> but it is better to know when it can fail.

Right. So, maybe I should add this as a test case.

Reply:
Satisfactory